File: sdks/thunkmetrc-py/client/src/thunkmetrc/client/services/units_of_measure.py
from typing import Any, Optional, Dict, List, Union
import urllib.parse

class UnitsOfMeasureClient:
    def __init__(self, client):
        self.client = client
    def get_active_units_of_measure(self) -> Any:
        """
        Retrieves all active units of measure.
        GET GetActiveUnitsOfMeasure
        Parameters:
        """
        query_params = {}
        path = f"/unitsofmeasure/v2/active"
        return self.client._send("GET", path, body, query_params)
    def get_inactive_units_of_measure(self) -> Any:
        """
        Retrieves all inactive units of measure.
        GET GetInactiveUnitsOfMeasure
        Parameters:
        """
        query_params = {}
        path = f"/unitsofmeasure/v2/inactive"
        return self.client._send("GET", path, body, query_params)

File: sdks/thunkmetrc-rs/client/src/services/units_of_measure.rs
use crate::client::MetrcClient;
use serde_json::Value;
use std::error::Error;

pub struct UnitsOfMeasureClient<'a> {
    pub(crate) client: &'a MetrcClient,
}

impl<'a> UnitsOfMeasureClient<'a> {
    /// GET GetActiveUnitsOfMeasure
    /// Retrieves all active units of measure.
    /// Parameters:
    pub async fn get_active_units_of_measure(&self, body: Option<&Value>) -> Result<Option<Value>, Box<dyn Error + Send + Sync>> {
        let path = format!("/unitsofmeasure/v2/active");

        self.client.send(reqwest::Method::GET, &path, body.map(|b| serde_json::to_value(b).unwrap()).as_ref()).await
    }
    /// GET GetInactiveUnitsOfMeasure
    /// Retrieves all inactive units of measure.
    /// Parameters:
    pub async fn get_inactive_units_of_measure(&self, body: Option<&Value>) -> Result<Option<Value>, Box<dyn Error + Send + Sync>> {
        let path = format!("/unitsofmeasure/v2/inactive");

        self.client.send(reqwest::Method::GET, &path, body.map(|b| serde_json::to_value(b).unwrap()).as_ref()).await
    }
}

